backend/users/serializers.py

```diff
@@ -1,8 +1,61 @@
+from django.contrib.auth import authenticate
+
 from rest_framework import serializers
+
 from .models import *
 
 
 # Serializers define the API representation
+class LoginSerializer(serializers.Serializer):
+    email = serializers.EmailField()
+    password = serializers.CharField()
+    
+    def validate_empty_values(self, data):
+        return super().validate_empty_values(data)
+    
+    def validate(self, data):
+        email = data.get('email')
+        password = data.get('password')            
+        
+        user = authenticate(request=self.context.get('request'),
+                            email=email, password=password)
+        
+        if not user:
+            msg ='Unable to log in with provided credentials.'
+            raise serializers.ValidationError(msg, code="authorization")
+        
+        data['user'] = user
+        return data
+    
+class RegisterSerializer(serializers.Serializer):
+    email = serializers.EmailField()
+    first_name = serializers.CharField()
+    last_name  = serializers.CharField()
+    password = serializers.CharField()
+    confirm_password = serializers.CharField()
+    
+    def validate_empty_values(self, data):
+        return super().validate_empty_values(data)
```

Comment:
this can be removed as it doesn't override anythin

Reply:
Okay , i will do that